### PR TITLE
Initial work on 0.0.2 alpha ui

### PR DESCRIPTION
**Pull request by loneil**
v0.0.2-alpha version of the GETOK UI

# Description

Add new components for user form entry of application configurations and JSON rendering of that.

https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-79
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-71

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (non-breaking change with enhancements to documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jujaga** [comment]
Test comment

File: frontend/src/components/ConfigGeneratedJson.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Vue.use(Clipboard);
+
+export default {
+  data() {
+    return {
+      snackbar: {
+        on: false,
+        text: "test",
+        color: ""
```
Comment (jujaga) [inline frontend/src/components/ConfigGeneratedJson.vue]: Seems a bit strange to have to specify an empty string for a color - might be better to have it have it be "info" instead as it should be the "default" color.

File: frontend/src/components/Header.vue [hunk 2]
```diff
@@ -12,129 +12,36 @@
...
-          link: '/groups'
-        },
-        {
-          title: 'About',
-          link: '/about'
-        }
-      ])
-    },
-  };
+export default {};
```
Comment (jujaga) [inline frontend/src/components/Header.vue]: I think we might be able to even omit this line and have it just be empty if we're effectively exporting nothing.

File: frontend/src/components/ManualConfig.vue [hunk 3]
```diff
@@ -0,0 +1,66 @@
+ <template>
```
Comment (jujaga) [inline frontend/src/components/ManualConfig.vue]: Please double check indentation spacing here - looks like it has one extra space.

File: frontend/src/components/ManualConfig.vue [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          );
+        })
+        .catch(function(error) {
+          console.error("Error:", error); // eslint-disable-line no-console
+          alert("ERROR, see console");
+        });
+    }
+  }
+};
+</script>
```
Comment (jujaga) [inline frontend/src/components/ManualConfig.vue]: Add a newline to end of file

File: frontend/src/store.js [hunk 5]
```diff
@@ -5,13 +5,129 @@ Vue.use(Vuex)
...
+              profileRoles: [
+                {
+                  applicationCode: newAppCfg.applicationAcronym,
+                  name: newAppCfg.applicationAcronym + "_ROLE"
+                },
+                {
+                  applicationCode: "CMSG",
+                  name: "SENDER"
+                }
+
```
Comment (jujaga) [inline frontend/src/store.js]: Extra whitespace line

File: frontend/src/components/ConfigGeneratedJson.vue (hunk 1 above)
Comment (loneil) [reply frontend/src/components/ConfigGeneratedJson.vue]: Good point. Info is the default

File: frontend/src/components/Header.vue (hunk 2 above)
Comment (loneil) [reply frontend/src/components/Header.vue]: Yup, works without it.

File: frontend/src/components/ManualConfig.vue (hunk 3 above)
Comment (loneil) [reply frontend/src/components/ManualConfig.vue]: Formatted.

File: frontend/src/components/ManualConfig.vue (hunk 4 above)
Comment (loneil) [reply frontend/src/components/ManualConfig.vue]: Thanks

File: frontend/src/store.js (hunk 5 above)
Comment (loneil) [reply frontend/src/store.js]: Formatted

**Pull request merged**